app/src/main/java/com/manage/ak/moneyreport/MainActivity.java

```diff
@@ -0,0 +1,555 @@
+
+                body = body.toLowerCase();
+
+                // some common keywords used in bank messages
+                if ((body.contains("debit") || body.contains("transaction") || body.contains("withdrawn")) && !body.contains("recharge") && !body.contains("paytm") && !body.contains("ola")) {
+                    t = "Personal Expenses";
+                } else if ((body.contains("credit") || body.contains("deposited")) && !body.contains("recharge") && !body.contains("paytm") && !body.contains("ola")) {
+                    t = "Income";
+                }
+
+                // switched according to the type to extract information from the message
+                switch (t) {
+                    case "Personal Expenses":
+                        sms.setMsgType(t);
+                        String a = getAmount(body);
+                        // getAmount is a method which gives the amount using pattern and matcher
+                        if (a != null) {
+                            sms.setMsgAmt(a);
+                            sms.setMsgBal(Double.toString(Double.parseDouble(BALANCE) - Double.parseDouble(a)));
+                            BALANCE = sms.getMsgBal();
+                            bankList.add(0, sms);
+                        } else {
+                            c.moveToPrevious();
+                            continue;
+                        }
+                        break;
+
+                    // for type of transaction income first the amount is extracted and then the balance is extracted
+                    case "Income":
+                        sms.setMsgType(t);
+                        String a1 = getAmount(body);
+                        if (a1 != null) {
+                            sms.setMsgAmt(a1);
+                            sms.setMsgBal(Double.toString(Double.parseDouble(BALANCE) + Double.parseDouble(a1)));
+                            BALANCE = sms.getMsgBal();
+                            bankList.add(0, sms);
+                        } else {
+                            c.moveToPrevious();
+                            continue;
+                        }
+                        break;
+                }
+                c.moveToPrevious();
+            }
+        } else {
+            // if no messages to read than a toast is displayed
+            Toast.makeText(MainActivity.this, "No sms to read!!", Toast.LENGTH_SHORT).show();
+        }
+        c.close();
+        if (bankList.size() > 0) {
+            bankBalance.setText("₹ " + bankList.get(0).getMsgBal());
+            estimateDate.setText(bankList.get(0).getFormatDate());
+        } else {
+            bankBalance.setText("₹ " + "0.0");
+            estimateDate.setText(" ");
+        }
+    }
+
+    // getting amount by matching the pattern
+    public String getAmount(String data) {
+        // pattern - rs. **,***.**
+        String pattern1 = "(inr)+[\\s]?+[0-9]*+[\\\\,]*+[0-9]*+[\\\\.][0-9]{2}";
```

Comment:
Your patterns here are not extensive which why  your app is only showing a few transactions.